File: Structural/Adapter/Java/MediaAdapter.java
// MediaAdapter.java
public class MediaAdapter implements MediaPlayer {
  private AdvancedMediaPlayer advancedMediaPlayer;

  public MediaAdapter(String audioType) {
    if (audioType.equalsIgnoreCase("MP4")) {
      advancedMediaPlayer = new MP4Player();
    } else if (audioType.equalsIgnoreCase("VLC")) {
      advancedMediaPlayer = new VLCPlayer();
    }
  }

  @Override
  public void play(String audioType, String fileName) {
    if (audioType.equalsIgnoreCase("MP4")) {
      advancedMediaPlayer.playMP4(fileName);
    } else if (audioType.equalsIgnoreCase("VLC")) {
      advancedMediaPlayer.playVLC(fileName);
    }
  }
}

File: Structural/Adapter/JavaScript/mediaAdapter.js
// mediaAdapter.js
const MediaPlayer = require("./mediaPlayer");
const MP4Player = require("./mp4Player");
const VLCPlayer = require("./vlcPlayer");

class MediaAdapter extends MediaPlayer {
  constructor(audioType) {
    super();
    if (audioType.toLowerCase() === "mp4") {
      this.advancedMediaPlayer = new MP4Player();
    } else if (audioType.toLowerCase() === "vlc") {
      this.advancedMediaPlayer = new VLCPlayer();
    } else {
      this.advancedMediaPlayer = null;
    }
  }

  play(audioType, fileName) {
    if (this.advancedMediaPlayer) {
      if (audioType.toLowerCase() === "mp4") {
        this.advancedMediaPlayer.playMP4(fileName);
      } else if (audioType.toLowerCase() === "vlc") {
        this.advancedMediaPlayer.playVLC(fileName);
      }
    }
  }
}

module.exports = MediaAdapter;
